programs/mango-v4/src/instructions/perp_liq_base_or_positive_pnl.rs

```diff
@@ -230,163 +238,294 @@ pub(crate) fn liquidation_action(
+    // 5. If perp_overall_weight>0, reduce base further while total health under threshold.
+
     // Take over the liqee's base in exchange for quote
     let liqee_base_lots = liqee_perp_position.base_position_lots();
-    // Each lot the base position gets closer to 0, the unweighted perp health contribution
-    // increases by this amount.
-    let unweighted_health_per_lot;
+
+    // Each lot the base position gets closer to 0, the "unweighted health unsettled pnl"
+    // increases by this amount
+    let uhupnl_per_lot;
+
     // -1 (liqee base lots decrease) or +1 (liqee base lots increase)
     let direction: i64;
+
     // Either 1+fee or 1-fee, depending on direction.
-    let fee_factor;
+    let base_fee_factor;
+
     if liqee_base_lots > 0 {
         require_msg!(
             max_base_transfer >= 0,
             "max_base_transfer can't be negative when liqee's base_position is positive"
         );
 
-        // the unweighted perp health contribution gets reduced by `base * price * perp_init_asset_weight`
-        // and increased by `base * price * (1 - liq_fee) * quote_init_asset_weight`
-        let quote_init_asset_weight = I80F48::ONE;
+        // the health_unsettled_pnl gets reduced by `base * base_price * perp_init_asset_weight`
+        // and increased by `base * base_price * (1 - liq_fee)`
         direction = -1;
-        fee_factor = I80F48::ONE - perp_market.base_liquidation_fee;
-        let asset_price = perp_info.prices.asset(HealthType::LiquidationEnd);
-        unweighted_health_per_lot =
-            -asset_price * base_lot_size * perp_market.init_base_asset_weight
-                + oracle_price_per_lot * quote_init_asset_weight * fee_factor;
+        base_fee_factor = I80F48::ONE - perp_market.base_liquidation_fee;
+        uhupnl_per_lot =
+            oracle_price_per_lot * (-perp_market.init_base_asset_weight + base_fee_factor);
     } else {
         // liqee_base_lots <= 0
         require_msg!(
             max_base_transfer <= 0,
             "max_base_transfer can't be positive when liqee's base_position is negative"
         );
 
-        // health gets increased by `base * price * perp_init_liab_weight`
-        // and reduced by `base * price * (1 + liq_fee) * quote_init_liab_weight`
-        let quote_init_liab_weight = I80F48::ONE;
+        // health gets increased by `base * base_price * perp_init_liab_weight`
+        // and reduced by `base * base_price * (1 + liq_fee)`
         direction = 1;
-        fee_factor = I80F48::ONE + perp_market.base_liquidation_fee;
-        let liab_price = perp_info.prices.liab(HealthType::LiquidationEnd);
-        unweighted_health_per_lot = liab_price * base_lot_size * perp_market.init_base_liab_weight
-            - oracle_price_per_lot * quote_init_liab_weight * fee_factor;
+        base_fee_factor = I80F48::ONE + perp_market.base_liquidation_fee;
+        uhupnl_per_lot =
```

Comment:
liab_price * base_lot_size is getting simplified to oracle_price_per_lot, is this intentional? same in the else clause